### ISSUE TITLE
Generating ANI svg is broken

### ISSUE THREAD

**Issue by TommyH-Tran** [eead-csic-compbio/get_homologues#98]
**I am trying to generate an ANI using the ANI .tab file. However, when I try to generate the .svg it seems to broken.**

**I am using the command:**
`./plot_matrix_heatmap.sh -i /Users/tommytran/Downloads/Cpr1274CJEI_f0_0taxa_CDS_algOMCL_e0_C90_Avg_identity.tab -d 2 -t "ALL CDS Cpr" -W 25 -H 30
`

**This is the output:** 

`
plot_matrix_heatmap.sh vv1.0.4_31Jan18 run started at 2022_05_21-14.27.42
        working directory : /Users/tommytran/get_homologues
        input tab_file : /Users/tommytran/Downloads/Cpr1274CJEI_f0_0taxa_CDS_algOMCL_e0_C90_Avg_identity.tab | sim_cutoff : 100 | max_decimals : 2
	subset_matrix : 0 | regex : 
        text:ALL CDS Cpr|margin_hor:18|margin_vert:18|points:15
	angle:45|charExp:1.0|right_margin:10
        width:25|height:30|outformat:svg
        reorder_clusters:1|remove_colnames:0|key_xaxis:Value|do_bioNJ:0
	k:
        
Plotting file /Users/tommytran/Downloads/Cpr1274CJEI_f0_0taxa_CDS_algOMCL_e0_C90_Avg_identity_heatmap.svg
Error in library(factoextra) : there is no package called ‘factoextra’
Calls: suppressPackageStartupMessages -> withCallingHandlers -> library
Execution halted
ERROR: file /Users/tommytran/Downloads/Cpr1274CJEI_f0_0taxa_CDS_algOMCL_e0_C90_Avg_identity_heatmap.svg was NOT produced.

You can try option -C or alternatively remove columns in the matrix.

file ANDg_meand_silhouette_width_statistic_plot.svg was produced

 ERROR: file ./ANDg_hc_plot_cut_at_mean_silhouette_width_k5.svg
./ANDg_hc_plot_cut_at_mean_silhouette_width_k4.svg
./ANDg_hc_plot_cut_at_mean_silhouette_width_k3.svg
./ANDg_hc_plot_cut_at_mean_silhouette_width_k2.svg was NOT produced!

rm: Rplots.pdf: No such file or directory
`

**I had a it working about a week ago, and then I tried to install the package `factoextra` since it said It was missing then I got this output:** 

---
> install.packages("factoextra")
--- Please select a CRAN mirror for use in this session ---
Secure CRAN mirrors 

 1: 0-Cloud [https]
 2: Australia (Canberra) [https]
 3: Australia (Melbourne 1) [https]
 4: Australia (Melbourne 2) [https]
 5: Australia (Perth) [https]
 6: Austria [https]
 7: Belgium (Brussels) [https]
 8: Brazil (PR) [https]
 9: Brazil (RJ) [https]
10: Brazil (SP 1) [https]
11: Brazil (SP 2) [https]
12: Bulgaria [https]
13: Canada (MB) [https]
14: Canada (ON 3) [https]
15: Chile (Santiago) [https]
16: China (Beijing 2) [https]
17: China (Beijing 3) [https]
18: China (Hefei) [https]
19: China (Hong Kong) [https]
20: China (Guangzhou) [https]
21: China (Lanzhou) [https]
22: China (Nanjing) [https]
23: China (Shanghai 2) [https]
24: China (Shenzhen) [https]
25: Colombia (Cali) [https]
26: Costa Rica [https]
27: Cyprus [https]
28: Czech Republic [https]
29: Denmark [https]
30: East Asia [https]
31: Ecuador (Cuenca) [https]
32: Ecuador (Quito) [https]
33: Estonia [https]
34: France (Lyon 1) [https]
35: France (Lyon 2) [https]
36: France (Marseille) [https]
37: France (Paris 1) [https]
38: Germany (Erlangen) [https]
39: Germany (Leipzig) [https]
40: Germany (Göttingen) [https]
41: Germany (Münster) [https]
42: Germany (Regensburg) [https]
43: Greece [https]
44: Hungary [https]
45: Iceland [https]
46: Indonesia (Jakarta) [https]
47: Iran (Mashhad) [https]
48: Italy (Milano) [https]
49: Italy (Padua) [https]
50: Japan (Tokyo) [https]
51: Korea (Gyeongsan-si) [https]
52: Korea (Seoul 1) [https]
53: Korea (Ulsan) [https]
54: Malaysia [https]
55: Mexico (Mexico City) [https]
56: Morocco [https]
57: Netherlands [https]
58: New Zealand [https]
59: Norway [https]
60: South Africa (Johannesburg) [https]
61: Spain (A Coruña) [https]
62: Spain (Madrid) [https]
63: Sweden (Borås) [https]
64: Sweden (Umeå) [https]
65: Switzerland [https]
66: Taiwan (Taipei) [https]
67: Turkey (Denizli) [https]
68: Turkey (Istanbul) [https]
69: Turkey (Mersin) [https]
70: UK (Bristol) [https]
71: UK (London 1) [https]
72: USA (IA) [https]
73: USA (KS) [https]
74: USA (MI) [https]
75: USA (OH) [https]
76: USA (OR) [https]
77: USA (TN) [https]
78: USA (TX 1) [https]
79: Uruguay [https]
80: (other mirrors)

Selection: 78
also installing the dependencies ‘nloptr’, ‘pbkrtest’, ‘lme4’, ‘car’, ‘rstatix’, ‘FactoMineR’, ‘ggpubr’

trying URL 'https://cran.microsoft.com/src/contrib/nloptr_2.0.2.tar.gz'
Content type 'application/octet-stream' length 2219995 bytes (2.1 MB)
downloaded 2.1 MB

trying URL 'https://cran.microsoft.com/src/contrib/pbkrtest_0.5.1.tar.gz'
Content type 'application/octet-stream' length 246327 bytes (240 KB)
downloaded 240 KB

trying URL 'https://cran.microsoft.com/src/contrib/lme4_1.1-29.tar.gz'
Content type 'application/octet-stream' length 3306026 bytes (3.2 MB)
downloaded 3.2 MB

trying URL 'https://cran.microsoft.com/src/contrib/car_3.0-13.tar.gz'
Content type 'application/octet-stream' length 494341 bytes (482 KB)
downloaded 482 KB

trying URL 'https://cran.microsoft.com/src/contrib/rstatix_0.7.0.tar.gz'
Content type 'application/octet-stream' length 402481 bytes (393 KB)
downloaded 393 KB

trying URL 'https://cran.microsoft.com/src/contrib/FactoMineR_2.4.tar.gz'
Content type 'application/octet-stream' length 2822993 bytes (2.7 MB)
downloaded 2.7 MB

trying URL 'https://cran.microsoft.com/src/contrib/ggpubr_0.4.0.tar.gz'
Content type 'application/octet-stream' length 2689307 bytes (2.6 MB)
downloaded 2.6 MB

trying URL 'https://cran.microsoft.com/src/contrib/factoextra_1.0.7.tar.gz'
Content type 'application/octet-stream' length 1386480 bytes (1.3 MB)
downloaded 1.3 MB

* installing *source* package ‘nloptr’ ...
** package ‘nloptr’ successfully unpacked and MD5 sums checked
** using staged installation
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C++... yes
checking whether clang++ -std=gnu++14 accepts -g... yes
checking for clang++ -std=gnu++14 option to enable C++11 features... none needed
checking how to run the C++ preprocessor... clang++ -std=gnu++14 -E
checking whether the compiler supports GNU C++... (cached) yes
checking whether clang++ -std=gnu++14 accepts -g... (cached) yes
checking for clang++ -std=gnu++14 option to enable C++11 features... (cached) none needed
checking for pkg-config... /usr/local/bin/pkg-config
checking if pkg-config knows NLopt... no
checking for cmake... no

------------------ CMAKE NOT FOUND --------------------

CMake was not found on the PATH. Please install CMake:

 - sudo yum install cmake          (Fedora/CentOS; inside a terminal)
 - sudo apt install cmake          (Debian/Ubuntu; inside a terminal).
 - sudo pacman -S cmake            (Arch Linux; inside a terminal).
 - sudo brew install cmake         (MacOS; inside a terminal with Homebrew)
 - sudo port install cmake         (MacOS; inside a terminal with MacPorts)

Alternatively install CMake from: <https://cmake.org/>

ERROR: configuration failed for package ‘nloptr’
* removing ‘/usr/local/Cellar/r/4.2.0/lib/R/library/nloptr’
ERROR: dependency ‘nloptr’ is not available for package ‘lme4’
* removing ‘/usr/local/Cellar/r/4.2.0/lib/R/library/lme4’
ERROR: dependency ‘lme4’ is not available for package ‘pbkrtest’
* removing ‘/usr/local/Cellar/r/4.2.0/lib/R/library/pbkrtest’
ERROR: dependencies ‘pbkrtest’, ‘lme4’ are not available for package ‘car’
* removing ‘/usr/local/Cellar/r/4.2.0/lib/R/library/car’
ERROR: dependency ‘car’ is not available for package ‘rstatix’
* removing ‘/usr/local/Cellar/r/4.2.0/lib/R/library/rstatix’
ERROR: dependency ‘car’ is not available for package ‘FactoMineR’
* removing ‘/usr/local/Cellar/r/4.2.0/lib/R/library/FactoMineR’
ERROR: dependency ‘rstatix’ is not available for package ‘ggpubr’
* removing ‘/usr/local/Cellar/r/4.2.0/lib/R/library/ggpubr’
ERROR: dependencies ‘FactoMineR’, ‘ggpubr’ are not available for package ‘factoextra’
* removing ‘/usr/local/Cellar/r/4.2.0/lib/R/library/factoextra’

The downloaded source packages are in
	‘/private/var/folders/_y/j9xb7x3s3c7cz22lldv6rmd40000gn/T/RtmpbHD5yG/downloaded_packages’
Updating HTML index of packages in '.Library'
Making 'packages.html' ... done
Warning messages:
1: In install.packages("factoextra") :
  installation of package ‘nloptr’ had non-zero exit status
2: In install.packages("factoextra") :
  installation of package ‘lme4’ had non-zero exit status
3: In install.packages("factoextra") :
  installation of package ‘pbkrtest’ had non-zero exit status
4: In install.packages("factoextra") :
  installation of package ‘car’ had non-zero exit status
5: In install.packages("factoextra") :
  installation of package ‘rstatix’ had non-zero exit status
6: In install.packages("factoextra") :
  installation of package ‘FactoMineR’ had non-zero exit status
7: In install.packages("factoextra") :
  installation of package ‘ggpubr’ had non-zero exit status
8: In install.packages("factoextra") :
  installation of package ‘factoextra’ had non-zero exit status
> 
---

**Comment by eead-csic-compbio** [comment]
Please try to install cmake as suggested in the output :

    CMake was not found on the PATH. Please install CMake:

    sudo yum install cmake (Fedora/CentOS; inside a terminal)
    sudo apt install cmake (Debian/Ubuntu; inside a terminal).
    sudo pacman -S cmake (Arch Linux; inside a terminal).
    sudo brew install cmake (MacOS; inside a terminal with Homebrew)
    sudo port install cmake (MacOS; inside a terminal with MacPorts)

**Comment by eead-csic-compbio** [comment]
Alternatively you can try the recently released package in bioconda, where the dependencies are already installed. See the instructions in the [manual](http://eead-csic-compbio.github.io/get_homologues/manual/manual.html#SECTION00030000000000000000)

**Comment by TommyH-Tran** [comment]
I did the cmake install using `brew install make`, however that did not completely fix the issue as "factoextra" package was still missing. I fixed this by trying to download it from a different region. There are 78 options and I choose `1: 0-Cloud [https]` instead. I guess the location I was choosing was having problems sourcing the "factoextra" package. The option I was choosing before was `78: USA (TX 1) [https]`. Guess that one is broken or something...

After, doing these two things I was able to get it working to generate the svg file. Thanks for the help!
